docs/DeveloperGuide.adoc

```diff
@@ -0,0 +1,665 @@
+*** Pros: It would simplify usability and implementation since there will only be 4 options to choose from.
+*** Cons: It would reduce the usability for the 2 examples provided above as users would not be able to make events repeat every 2 weeks
+or 10 days, forcing them to have to manually type in the same event for as many times as it will occur if they wish to still keep track
+of that event.
+
+* Keeping `RepeatEvent` as a single entity within the list and not repeatedly adding new tasks with progressive dates when `RepeatCommand` is used.
+** Rationale: +
+It allows the repeated events to be easily removed or un-repeated as a there will only be a single `RepeatEvent` present in the list.
+** Alternative considered: +
+1. Repeatedly add new events with changes in dates for a fixed amount when repeat command is used.
+*** Pros: It will be simpler to implement and test if repeating events can be treated like any other events as coupling is lower.
+*** Cons: Deleting a repeating event would be difficult as there would be multiple entries to delete. It will also flood the user's list
+and increase the size of the local file that stores the `TaskList`.
+
+=== Edit Task Feature
+==== Implementation
+The `EditCommand` class extends the `Command` class by providing functions to edit specific tasks in the list of
+*ATAS*.
+
+Given below is an example usage scenario of the `edit` command.
+
+*Step 1* +
+The user types in `edit 1`. The `parseCommand()` method of the `Parser` class is called to obtain `edit` which is the type
+of command the user is entering.
+
+[WARNING]
+An `IncorrectCommand` class will be returned and an `UNKNOWN_COMMAND_ERROR` string from the `Messages` class will be passed
+into the constructor of that class if the command supplied was invalid.
+
+*Step 2* +
+The `parseCommand()` method subsequently calls the `prepareEditCommand()` method inside the same `Parser` class. This method
+splits the `fullCommand` string parameters into 2 tokens. The integer `1` will be obtained as the *Index* of the task
+specified in the list. This method returns a new instance of the `EditCommand` class, passing the integer `1` as the parameter.
+
+[WARNING]
+An `IncorrectCommand` class will be returned and a `NUM_FORMAT_ERROR` string from the `Messages` class will be passed
+into the constructor of that class if the number supplied was not an *integer*. +
+An `IncorrectCommand` class will be returned and a `INCORRECT_ARGUMENT_ERROR` string from the `Messages` class will be passed
+into the constructor of that class if there are no task index supplied by the user. +
+
+*Step 3* +
+A new instance of the `EditCommand` class is returned to the main method of *ATAS* with parameter `1` as described above.
+The `execute()` method of the `EditCommand` class is now called.
+
+*Step 4* +
+The `execute()` method in the `EditCommand` class first gets an input from the user on the details of the edited task.
+
+[TIP]
```

Comment:
Maybe the add feature / the parameters can be mentioned earlier / elaborate the adding function? 
Reason: to avoid potential confusion. Developer may be confused that add function is a sub function in Edit function